src/dom/accessibility.ts

```diff
@@ -1,23 +1,28 @@
-// TODO: Before exposing this, it might be worth making `prefersReducedMotion`
-// a `MotionValue` with a selectively-exposed API, for instance `get` and `onChange`.
+import { motionValue } from "../value"
 
-let prefersReducedMotion: boolean | null = null
+// Does this device prefer reduced motion? Returns `null` server-side.
+export const prefersReducedMotion = motionValue<boolean | null>(null)
 
 if (typeof window !== "undefined") {
-    const motionMediaQuery = window.matchMedia("(prefers-reduced-motion)")
+    if (window.matchMedia) {
+        const motionMediaQuery = window.matchMedia("(prefers-reduced-motion)")
 
-    const setReducedMotionPreferences = () =>
-        (prefersReducedMotion = motionMediaQuery.matches)
+        const setReducedMotionPreferences = () =>
+            prefersReducedMotion.set(motionMediaQuery.matches)
 
-    motionMediaQuery.addListener(setReducedMotionPreferences)
-    setReducedMotionPreferences()
+        motionMediaQuery.addListener(setReducedMotionPreferences)
+
+        setReducedMotionPreferences()
+    } else {
+        prefersReducedMotion.set(false)
+    }
 }
 
-/**
- * Determine whether the user prefers reduced motion.
- *
- * Returns `null` if called server-side.
- *
- * @internal
- */
-export const isReducedMotion = () => prefersReducedMotion
+export function determineShouldReduceMotion(
+    prefersReduced: boolean | null,
+    isReducedMotion: boolean | undefined
+): boolean {
+    return isReducedMotion !== undefined
+        ? isReducedMotion
+        : Boolean(prefersReduced)
```

Comment:
A nit here, but it feels like this would be clearer written as `typeof isReducedMotion === "boolean" ?`.